interfaces/backends/backends.go

```diff
@@ -28,21 +31,44 @@ import (
 	"github.com/snapcore/snapd/interfaces/seccomp"
 	"github.com/snapcore/snapd/interfaces/systemd"
 	"github.com/snapcore/snapd/interfaces/udev"
-	"github.com/snapcore/snapd/release"
 )
 
-// append when a new security backend is added
-var All = []interfaces.SecurityBackend{
-	&systemd.Backend{},
-	&seccomp.Backend{},
-	&dbus.Backend{},
-	&udev.Backend{},
-	&mount.Backend{},
-	&kmod.Backend{},
-}
+var All []interfaces.SecurityBackend = backends()
+
+func backends() []interfaces.SecurityBackend {
+	all := []interfaces.SecurityBackend{
+		&systemd.Backend{},
+		&seccomp.Backend{},
+		&dbus.Backend{},
+		&udev.Backend{},
+		&mount.Backend{},
+		&kmod.Backend{},
+	}
+	featureLevel, summary := aa.ProbeKernel().Evaluate()
+
+	// This should be logger.Noticef but due to ordering of initialization
+	// calls, the logger is not ready at this point yet and the message goes
+	// nowhere. Per advice from other snapd developers, we just print it
+	// directly.
+	//
+	// TODO: on this should become a user-visible message via the user-warning
+	// framework, so that users are aware that we have non-strict confinement.
+	// By printing this directly we ensure it will end up the journal for the
+	// snapd.service. This aspect should be retained even after the switch to
+	// user-warning.
+	if featureLevel != aa.Full {
+		fmt.Printf("WARNING: %s\n", summary)
```

Comment:
We cannot log the warning to the real log because of initialization order of our logger. This does the right thing because stdout is properly logged into the journal.